### PR TITLE
Added lint for TryFrom for checked integer conversion.

### PR DESCRIPTION
**Pull request by pJunger**
works towards #3947 

Added lint for try_from for checked integer conversion.
Should recognize simple & straight-forward checked integer conversions.

### CODE REVIEW

File: tests/ui/checked_conversions.stderr [hunk 1]
```diff
@@ -0,0 +1,52 @@
+error: Checked cast can be simplified: `u32::try_from`
+  --> $DIR/checked_conversions.rs:9:8
+   |
+LL |     if value <= (u32::max_value() as i64) && value >= 0 {
```
Comment (oli-obk) [inline tests/ui/checked_conversions.stderr]: Can you try creating a suggestion that will produce `u32::try_from(value).is_ok()`? You can change the test to be a rustfix test which will automatically check if the code still compiles after applying all suggestions.

Comment (pJunger) [reply tests/ui/checked_conversions.stderr]: I added a new commit that should do exactly that - although I am not sure how to add a (potentially missing) include for std::convert::TryFrom.

**Comment by pJunger** [comment]
No worries!
I agree that this is only a partial solution - though I guess it's the part that's more error prone (and easier to implement).
Should I remove the 'Closes' comment?

**Comment by oli-obk** [comment]
> Should I remove the 'Closes' comment?

yea, we'll keep the issue open for the full fix

File: tests/ui/checked_conversions.stderr (hunk 1 above)
Comment (oli-obk) [reply tests/ui/checked_conversions.stderr]: There's no need for an include, the compiler will tell the user what they need if they need it.

**Comment by oli-obk** [comment]
you have some CI failures. Most likely you need to rebase and use the latest master branch of rustc.

File: tests/ui/checked_conversions.stderr (hunk 1 above)
Comment (pJunger) [reply tests/ui/checked_conversions.stderr]: But it is still needed for the rustfix test to work, right? Or am I missing something there, too?

**Comment by oli-obk** [comment]
@bors r+

Thanks!

**Comment by pJunger** [comment]
@oli-obk Could we try again? (I assume the last PR fixed the testsuite issue?)

**Comment by oli-obk** [comment]
@bors r+

sorry about the symbol churn.

**Comment by pJunger** [comment]
Sure! I'm glad it worked out.

**Pull request merged**